cadc-test-vosi/src/main/java/ca/nrc/cadc/vosi/AvailabilityTest.java

```diff
@@ -1,137 +1,173 @@
             String resourceIdentifierName = AvailabilityTest.class.getName() + ".resourceIdentifier";
             String resourceIdentifierValue = System.getProperty(resourceIdentifierName);
             log.debug(resourceIdentifierName + "=" + resourceIdentifierValue);
             this.resourceIdentifier = URI.create(resourceIdentifierValue);
-        }
-        else
+        } else {
             this.resourceIdentifier = resourceIdentifier;
+        }
     }
 
     @Test
-    public void testAvailability()
-    {
-        RegistryClient rc = new RegistryClient();
-        try
-        {
-            RegistryClient regClient = new RegistryClient();
-            URL availabilityURL = regClient.getServiceURL(resourceIdentifier, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
-            log.info("availability url: " + availabilityURL);
-
-            CheckWebService checkWebService = new CheckWebService(availabilityURL.toString());
+    public void testAvailability() {
+        try {
+            CheckWebService checkWebService = new CheckWebService(lookupServiceURL().toString());
             checkWebService.check();
-        }
-        catch (Throwable t)
-        {
+        } catch (Throwable t) {
             log.error("unexpected exception", t);
             Assert.fail("unexpected exception: " + t);
         }
     }
+
+    @Test
+    public void testClientIP() throws Exception {
+        final URL serviceURL = lookupServiceURL();
+        final AvailabilityClient availabilityClient = new AvailabilityClient();
+        final ByteArrayOutputStream outputStream = availabilityClient.getOutputStream();
+        final HttpDownload download = availabilityClient.getHttpDownload(serviceURL, outputStream);
+        download.run();
+
+        final Document xml =
+            XmlUtil.buildDocument(outputStream.toString("UTF-8"),
+                                  VOSI.AVAILABILITY_NS_URI, VOSI.AVAILABILITY_SCHEMA);
+        Assert.assertTrue("Should have comment.", hasClientIPComment(xml.getRootElement()));
+    }
+
+    URL lookupServiceURL() {
+        RegistryClient regClient = new RegistryClient();
+        URL availabilityURL = regClient.getServiceURL(resourceIdentifier, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
+        log.info("availability url: " + availabilityURL);
+
+        return availabilityURL;
+    }
+
+    /**
+     * Obtain whether the given Document contains a Comment containing the given string.
+     *
+     * @param element The element to check.
+     * @return True if one or more comments match, or False otherwise.
+     */
+    boolean hasClientIPComment(final Element element) {
+        final List<Content> comments = element.getContent(new ContentFilter(ContentFilter.COMMENT) {
+            @Override
+            public Content filter(final Object obj) {
+                final Content c = super.filter(obj);
+                return ((c != null) && c.getValue().contains("<clientip>")) ? c : null;
```

Comment:
You'll need to expand this test so that the IP address of the host that is running the integration test is actually the one that is reported in the comment of the availability document.